components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/interactive_main.cc

```diff
@@ -0,0 +1,141 @@
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/interactive_main.h"
+
+#include <wrl/client.h>
+
+#include <memory>
+#include <utility>
+
+#include "base/logging.h"
+#include "base/run_loop.h"
+#include "base/task/single_thread_task_executor.h"
+#include "base/task/thread_pool/thread_pool_instance.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/common/service_constants.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/common/wireguard_utils.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/brave_vpn_interactive_strings_en.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/brave_vpn_tray_command_ids.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/interactive_utils.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/resources/resource.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/status_icon_win.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/interactive/status_tray_win.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace brave_vpn {
+
+namespace {
+
+constexpr char kBraveAccountURL[] = "http://account.brave.com/";
+constexpr char kAboutBraveVPNURL[] = "https://brave.com/firewall-vpn/";
+
+std::u16string GetStatusIconTooltip(bool connected, bool error) {
+  if (error) {
+    return brave::kBraveVpnIconTooltipError;
+  }
+  return connected ? brave::kBraveVpnIconTooltipConnected
+                   : brave::kBraveVpnIconTooltip;
+}
+
+gfx::ImageSkia GetStatusTrayIcon(bool connected, bool error) {
+  bool dark_theme = UseDarkTheme();
+  if (error) {
+    int status_icon_id = dark_theme ? IDR_BRAVE_VPN_TRAY_LIGHT_ERROR
+                                    : IDR_BRAVE_VPN_TRAY_DARK_ERROR;
+    return GetIconFromResources(status_icon_id, {64, 64});
+  }
+  int light_icon_id =
+      connected ? IDR_BRAVE_VPN_TRAY_LIGHT_CONNECTED : IDR_BRAVE_VPN_TRAY_LIGHT;
+  int dark_icon_id =
+      connected ? IDR_BRAVE_VPN_TRAY_DARK_CONNECTED : IDR_BRAVE_VPN_TRAY_DARK;
+  return GetIconFromResources(dark_theme ? light_icon_id : dark_icon_id,
+                              {64, 64});
+}
+
+}  // namespace
+
+InteractiveMain* InteractiveMain::GetInstance() {
+  static base::NoDestructor<InteractiveMain> instance;
+  return instance.get();
+}
+
+InteractiveMain::InteractiveMain() = default;
+
+InteractiveMain::~InteractiveMain() = default;
+
+void InteractiveMain::SetupStatusIcon() {
```

Comment:
why is this called `*Main`? That convention is normally used for `main` or equivalent entry point for an executable?